### PR TITLE
Add additional hash for Vietnamese Calendar

### PR DESCRIPTION
**Pull request by jonathanpike**
Further to our discussion on [in the Definitions repo](https://github.com/holidays/definitions/pull/28), I've added an additional regional lunar hash for Vietnam.  I've added test data for all of the holiday data that I can get my hands on, and it passes.  I also added commentary to `lunar_date.rb` to make it more approachable.

### CODE REVIEW

File: lib/holidays/date_calculator/lunar_date.rb [hunk 1]
```diff
@@ -188,10 +341,18 @@ def to_s
       ].freeze
 
       MAX_YEAR_NUMBER = 150
+
+      # Given the region, CALENDAR_YEAR_INFO_MAP looks up the date 
+      # table and uses it in the calculation
       CALENDAR_YEAR_INFO_MAP = {
-        ko: KOREAN_LUNAR_YEAR_INFO
+        ko: KOREAN_LUNAR_YEAR_INFO,
```
Comment (ppeble) [inline lib/holidays/date_calculator/lunar_date.rb]: Could we switch to `:kr` so the symbols match with the definitions? What do you think?

**Comment by jonathanpike** [comment]
Thanks for the feedback @ppeble. In regards to method arguments, couldn't we add "region" to the list of valid arguments, and then pass that in to the lunar_to_solar method? That seems to be an easy fix.

Also, I'll change the symbol to :kr. I was thinking about doing that, anyways. I'll update the pr.

**Comment by jonathanpike** [comment]
After looking at the file that you referenced (custom method validator), I'm a little confused. It seems to me that that code refers to the custom methods defined in the definition files and then parsed into Ruby methods in the generated definition files. Since lunar_to_solar is a default, provided by the library method, does it have the same restrictions?

**Comment by ppeble** [comment]
@jonathanpike Yep! That is a valid solution for the short term. I'll do that this week. We can do that as part of a separate PR on my side.

I'm not terribly happy with the restrictions I put in place but like I said, at the time there was no reason to make it more complex. I'm definitely open to making it more flexible.

File: lib/holidays/date_calculator/lunar_date.rb [hunk 2]
```diff
@@ -3,9 +3,9 @@ module DateCalculator
     # Copied from https://github.com/sunsidew/ruby_lunardate
     # Graciously allowed by JeeWoong Yang (https://github.com/sunsidew)
     class LunarDate
-      attr_accessor :year, :month, :day, :is_leap_month
+      attr_accessor :year, :month, :day
 
-      def to_solar(year, month, day, is_leap_month = false, calendar_symbol = :ko)
+      def to_solar(year, month, day, calendar_symbol)
```
Comment (ppeble) [inline lib/holidays/date_calculator/lunar_date.rb]: Unfortunately the argument name _must_ match the exact expected name, in this case `region`. 😦 

Could we change it here? Then I will merge and do whatever necessary updates to make it all work (along with the other PR in `definitions`).

**Comment by jonathanpike** [comment]
@ppeble I've updated both this PR and the definitions PR.  One thing I didn't do was verify that the received parameter was a symbol -- can I expect one or should I do the conversion in the lunar_date class?

**Comment by ppeble** [comment]
@jonathanpike Um....it's probably safer to verify. I am pretty sure it will remain a symbol all the way through but to be safe we should check.

I hope you don't mind, I'm going to merge master back in and merge this. If there are any lingering effects I will handle them on my end. Thanks for being patient through this!

**Pull request merged**